### ISSUE TITLE
Add field selector support to hostSelector

### ISSUE THREAD

**Issue by russellb** [metal3-io/cluster-api-provider-baremetal#81]
PR #74 added the `hostSelector` field to the baremetal Machine provider spec.  (See doc/api.md)

The initial `hostSelector` support includes the ability to specify `BareMetalHost` matching criteria based on matching labels.  It would also be interesting to add the option of specifying field selectors, so we can match on more than just labels.

A simple use case would be to match on `metadata.name` if you wanted to specify a specific `BareMetalHost` by name.  A workaround for this use case would be to add a unique label to each host.

Perhaps there are more interesting use cases around matching on hardware inventory details that are in the `BareMetalHost` status.

**Comment by alanmeadows** [comment]
@russellb totally agree on the use case of a specific host by name and the need for this with regards to other match criteria, but for posterity but couldn't you use labels to match hosts by name with `hostSelector` as it is implemented today? In other words, match label `kubernetes.io/hostname=myhostname` for a specific host which should be automatically created out of the box in standard clusters. 

Not suggesting this ability makes this feature request any less needed - just wanted to check on thoughts on using that approach for host targeting.

**Comment by russellb** [comment]
Yes, for sure, you could add labels for as much information as you want and use it as implemented.  We could also automate rolling up some key hardware information as labels.  It's not clear to me yet how useful the field selector really is, but seemed worth having on the backlog to play around with.  :-)

**Comment by dhellmann** [comment]
I think we're going to want to do both, roll some data up to labels and allow the field selector. We're going to have scalars like hardware profile names and RAM that are easy to put into labels. We also have other discovered hardware info that may or may not fit as labels, but that could be used by the field selector.
